### DISCUSSION TITLE
support Valetudo now?

### DISCUSSION THREAD

**Discussion by fineemb** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#236] [Q&A]
Valetudo also offers a camera.map_data
![image](https://user-images.githubusercontent.com/16739914/144703437-ccd2edd2-6b6f-4863-81b3-28fe47f30b03.png)

**Comment by fineemb** [comment]
![image](https://user-images.githubusercontent.com/16739914/144703582-bd79c8ff-58f3-4209-bc66-5432a8a566b4.png)

**Comment by PiotrMachowski** [comment]
How the camera is created?

**Reply by dugite-code** [reply]
The map is generated on the fly by the map card (from the raw MQTT data): https://github.com/TheLastProject/lovelace-valetudo-map-card/blob/master/valetudo-map-card.js#L181

If you wanted to use the camera entity from the MQTT you would have to generate the map on the fly as well.

Another example of generating a map from the Valetudo MQTT info: https://github.com/alexkn/node-red-contrib-valetudo

**Reply by PiotrMachowski** [reply]
@dugite-code I'd really prefer not to do it. [I can't believe it's not Valetudo](https://github.com/Hypfer/Icantbelieveitsnotvaletudo) seems a way to go instead
